rfcs/text/0030/observer.yml

```diff
@@ -0,0 +1,168 @@
+      level: core
+      type: keyword
+      description: >
+        Hostname of the observer.
+
+    - name: name
+      level: extended
+      type: keyword
+      short: Custom name of the observer.
+      description: >
+        Custom name of the observer.
+
+        This is a name that can be given to an observer. This can be helpful
+        for example if multiple firewalls of the same model are used in an
+        organization.
+
+        If no custom name is needed, the field can be left empty.
+
+      example: 1_proxySG
+
+    - name: product
+      level: extended
+      type: keyword
+      description: >
+        The product name of the observer.
+      example: s200
+
+    - name: vendor
+      level: core
+      type: keyword
+      description: >
+        Vendor name of the observer.
+      example: Symantec
+
+    - name: version
+      level: core
+      type: keyword
+      description: >
+        Observer version.
+
+    - name: serial_number
+      level: extended
+      type: keyword
+      description: >
+        Observer serial number.
+
+    - name: type
+      level: core
+      type: keyword
+      short: The type of the observer the data is coming from.
+      description: >
+        The type of the observer the data is coming from.
+
+        There is no predefined list of observer types. Some examples are
+        `forwarder`, `firewall`, `ids`, `ips`, `proxy`, `poller`, `sensor`, `APM server`.
+      example: firewall
+
+    - name: ingress
+      level: extended
+      type: object
+      short: Object field for ingress information
+      description: >
+        Observer.ingress holds information like interface number and name, vlan, and zone information to
+        classify ingress traffic.  Single armed monitoring such as a network sensor on a span port should
+        only use observer.ingress to categorize traffic.
+
+    - name: ingress.zone
+      level: extended
+      type: keyword
+      short: Observer ingress zone
+      example: DMZ
+      description: >
+        Network zone of incoming traffic as reported by the observer to categorize the source area of ingress
+        traffic. e.g. internal, External, DMZ, HR, Legal, etc.
+
+    - name: egress
+      level: extended
+      type: object
+      short: Object field for egress information
+      description: >
+        Observer.egress holds information like interface number and name, vlan, and zone information to
+        classify egress traffic.  Single armed monitoring such as a network sensor on a span port should
+        only use observer.ingress to categorize traffic.
+
+    - name: egress.zone
+      level: extended
+      type: keyword
+      short: Observer Egress zone
+      example: Public_Internet
+      description: >
+        Network zone of outbound traffic as reported by the observer to categorize the destination area of egress
+        traffic, e.g. Internal, External, DMZ, HR, Legal, etc.
+
+    - name: boot.id
```

Comment:
boot.it and pid_ns_ino are the only additions to observer, and will serve as a seed for unique process ids (entity_id, or a new process.id (tbd) )

Reply:
I'm not sure these two fields are the best fit for `observer.*`. What do you think about instead adding them to [`os.yml`](https://github.com/elastic/ecs/blob/main/schemas/os.yml) or maybe [`host.yml`](https://github.com/elastic/ecs/blob/main/schemas/host.yml) ?

Reply:
Yea, @ferullo suggested they could be added to "process". I chose observer because @m-sample 's domain model doc put these in an 'observer' category (w/o prior knowledge of the 'observer' fieldset)

boot.id kind makes sense to live under "host" as it's the boot_id for the system. (though it will change with host restarts). pid_ns_ino probably makes sense to live in "process" as it's pid namespace centric and less tied to the host.

Reply:
These 2 fields are effectively defining the point of observation/perspective from which the process events are observed and identified.  Maintaining this perspective information is important to identifying these processes uniquely over time (for events stored in ES).

pid_ns_ino identifies an observation perspective on a running Linux server.    If someone accidentally or mistakenly installs an agent in a pid-namespaced container this is important, especially if there is another agent installed on the same server.

boot_id identifies a specific boot of a Linux server and helps make the specific process unique over time (even when the pid re-used).

Reply:
Both fields are moved to 'host' fieldset now. will resolve this convo.